app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -46,6 +46,7 @@ def login_shop
       sess = ShopifyAPI::Session.new(shop_name, token)
       session[:shopify] = ShopifyApp::SessionRepository.store(sess)
       session[:shopify_domain] = shop_name
+      session[:shopify_user] = associated_user if associated_user.present?
```

Comment:
While I'm looking at this... I believe this code path should be protected against [session fixation](https://www.owasp.org/index.php/Session_fixation), this can be done with this:
```
    session.options[:renew] = true
    session.delete(:_csrf_token)
```